azure-cli/ad/Add-OAuthUserImpersonationScope.ps1

```diff
@@ -0,0 +1,49 @@
+function Add-OAuthUserImpersonationScope(
+    [Parameter(Mandatory = $true)]
+    [string] $spnId
+) {
+  # import utility functions
+  . "$PSScriptRoot\..\utilities\deploy.naming.ps1"
+
+  $existingManifest = az rest `
+    --method Get `
+    --url https://graph.microsoft.com/v1.0/applications/$spnId `
+    --headers "Content-Type=application/json" `
+    | ConvertFrom-Json
+
+  Throw-WhenError -output $existingManifest
+
+  $existingPermissionScopes = $existingManifest.api.oauth2PermissionScopes
+  $existingScope = $existingPermissionScopes | Where-Object {$_.value -eq "user_impersonation"}
+
+  if ($null -eq $existingScope){
+    $newScope = @{
+      adminConsentDescription = "Allow the application to access $($existingManifest.displayName) on behalf of the signed-in user."
+      adminConsentDisplayName = "Access $($existingManifest.displayName)"
+      id = $([System.guid]::NewGuid().toString())
+      isEnabled = $true
+      type = "User"
+      userConsentDescription = "Allow the application to access $($existingManifest.displayName) on your behalf."
+      userConsentDisplayName = "Access $($existingManifest.displayName)"
+      value = "user_impersonation"
+    }
+
+    $existingPermissionScopes += $newScope
```

Comment:
should this be moved inside the `if ($null -eq $existingScope) { ... }`  scope ?